src/app/content/highlights/hooks/addCurrentPageToSummaryFilters.ts

```diff
@@ -0,0 +1,35 @@
+import { ActionHookBody, AppServices, MiddlewareAPI,  } from '../../../types';
+import { actionHook } from '../../../utils';
+import { bookAndPage } from '../../selectors';
+import { findArchiveTreeNode } from '../../utils/archiveTreeUtils';
+import { stripIdVersion } from '../../utils/idUtils';
+import { addCurrentPageToSummaryFilters, filtersChange, setChaptersFilter } from '../actions';
+import * as select from '../selectors';
+
+export const hookBody: ActionHookBody<typeof addCurrentPageToSummaryFilters> = ({
+  dispatch, getState,
+}: MiddlewareAPI & AppServices) => async() => {
+  const state = getState();
+  const {book, page} = bookAndPage(state);
+  const selectedChapters = select.chaptersFilter(state);
+  if (!book || !page || typeof(window) === 'undefined') {
+    return;
+  }
+
+  // Add current page or their parent to the chaptersFilter
+  const treeNode = findArchiveTreeNode(book.tree, page.id);
+  if (!treeNode) { return; }
+  const nodeParent = treeNode.parent!;
```

Comment:
chapters aren't always one level deep, sometimes they have groups of pages inside them. i'd stick to iterating over the chapters and checking to see if the current page id is any descendant of them.

Reply:
Ok, now I'm iterating over sections